src/terminology.md

```diff
@@ -8,38 +8,54 @@ We group terms into related categories as follows:
-- *Wallet Viewer*: a component which provides a view into the history of funds for particular addresses, given appropriate *viewing keys*. This history may include both transparent and private details.
-- *Wallet Spender*: a component that enables generating new transactions which transfer funds to recipients.
+<span id="definition-validator"></span>**Validator**: a component which locally verifies the correctness of consensus. This includes verifying that the local view of chain history matches consensus requirements, encompassing block tip selection, block validity, and transaction validity.[^validator-distinction]
+
+<span id="definition-block-proposer"></span>**Block Proposer**: a component which proposes a block of transactions to the network. If accepted by network consensus, this block extends the consensus state of the ledger.
+
+<span id="definition-pow-proposer"></span>**PoW Proposer**: a Block Proposer which uses PoW as the proposal mechanism. In NU5 and PoW+TFL, the only Block Proposers are PoW Proposers. In practice, PoW Proposers are typically mining pools, although a solo miner is also a PoW Proposer. We use this term to be more precise than the common term "miner" which can conflate this role with the following:
+
+<span id="definition-pow-hashrate-provider"></span>**PoW Hashrate Provider**: a component which contributes mining resources towards PoW block proposals. In practice, mining pools rely on a userbase of Hashrate Providers to scale their operation, and solo miners have this capacity "in-house".
+
+<span id="definition-block-finalizer"></span>**Block Finalizer**: a component which contributes to consensus progress on the *finality* of a block.
+
+<span id="definition-wallet-viewer"></span>**Wallet Viewer**: a component which provides a view into the history of funds for particular addresses, given appropriate *viewing keys*. This history may include both transparent and private details.
+
+<span id="definition-wallet-spender"></span>**Wallet Spender**: a component that enables generating new transactions which transfer funds to recipients.
 
 ## Blockchain State
 
-- *Transaction*: a modification of the ledger, issued (by definition) by a Wallet Spender. A transaction cannot become part of the consensus ledger unless all Validators would accept it as valid according to *Transaction Validity Rules*.
-- *Block*: **\[TODO\]**
-- *Block History*: **\[TODO\]** _(nodes can see multiple local histories and select one as canonical according to consensus)
-- *Pending Blocks*: **\[TODO\]**
-- *Pending Transactions*: **\[TODO\]**
-- *Final Blocks*: **\[TODO\]**
-- *Final Transactions*: **\[TODO\]**
+
+<span id="definition-transaction"></span>**Transaction**: a modification of the ledger, issued (by definition) by a Wallet Spender. A transaction cannot become part of the consensus ledger unless all Validators would accept it as valid according to *Transaction Validity Rules*.
```

Comment:
Transactions are only valid or invalid in a context (for Crosslink, a bc-context).